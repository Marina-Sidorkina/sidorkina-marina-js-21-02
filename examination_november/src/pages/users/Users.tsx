import React from 'react';
import './Users.scss';
import Main from '../../components/main/Main';
import UsersList from '../../components/usersList/UsersList';
import Paginator from '../../components/paginator/Paginator';

const Users = () => (
  <Main>
    <UsersList />
    <Paginator />
  </Main>
);

export default Users;
