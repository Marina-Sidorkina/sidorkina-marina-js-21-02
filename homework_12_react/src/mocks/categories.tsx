export const categories = [
  {
    id: "promo-1",
    title: "Пианино",
    message: "Выбор для учебы",
    href: "#"
  },
  {
    id: "promo-2",
    title: "Рояли",
    message: "Выбор профессионала",
    href: "#"
  }
];
