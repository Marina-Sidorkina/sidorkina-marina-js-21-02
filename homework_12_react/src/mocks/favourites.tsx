import schK280 from './img/schimmel-k-280.jpg';
import fazF156 from './img/fazioli-f-156.png';
import yamYus1 from './img/yamaha-yus-1.png';
import kawNd21 from './img/kawai-nd-21.jpg';
import schK256 from './img/schimmel-k-256.jpg';
import fazF183 from './img/fazioli-f-183.png';

export const favourites = [
  {
    key: "schK280",
    name: "РОЯЛЬ SCHIMMEL K 280",
    src: schK280,
    href: "#"
  },
  {
    key: "fazF156",
    name: "РОЯЛЬ FAZIOLI F156",
    src: fazF156,
    href: "#"
  },
  {
    key: "yamYus1",
    name: "ПИАНИНО YAMAHA YUS1",
    src: yamYus1,
    href: "#"
  },
  {
    key: "kawNd21",
    name: "ПИАНИНО KAWAI ND-21",
    src: kawNd21,
    href: "#"
  },
  {
    key: "schK256",
    name: "РОЯЛЬ SCHIMMEL K 256",
    src: schK256,
    href: "#"
  },
  {
    key: "fazF183",
    name: "РОЯЛЬ FAZIOLI F183",
    src: fazF183,
    href: "#"
  }
];
