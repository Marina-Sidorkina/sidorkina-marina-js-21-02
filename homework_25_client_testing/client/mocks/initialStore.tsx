const initialStore = {
  userModal: {
    isOpened: false,
    values: {
      picture: '',
      name: 'John Doe',
      gender: 'male',
      dateOfBirth: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      registerDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC',
      },
      id: '12345',
      email: 'test',
      phone: '55555'
    }
  },
  login: {
    data: {
      error: false,
      isLoading: false,
      inputValue: '',
      authorizedUserId: '123345',
      authorizedUserName: 'Name',
      authorizedUserPicture: ''
    }
  },
  registration: {
    values: {
      name: '',
      gender: '',
      dateOfBirth: '',
      email: '',
      phone: '',
    },
    isLoading: false,
    error: false
  },
  languageSelector: {
    value: 'ru'
  },
  postModal: {
    isOpened: false,
    currentPostId: '12345'
  },
  userPosts: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  postsList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      posts: [{
        id: '12345',
        image: '',
        text: 'text',
        publishDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC'
        },
        owner: {
          firstName: 'John',
          lastName: 'Doe',
          picture: '',
          title: 'mr',
          id: '12345'
        }
      }]
    }
  },
  usersList: {
    data: {
      page: 0,
      total: 100,
      perPage: 6,
      users: [{
        firstName: 'John',
        lastName: 'Doe',
        title: 'mr',
        id: '12345'
      }]
    }
  },
  userInfo: {
    data: {
      isLoading: false,
      user: {
        firstName: 'John',
        lastName: 'Doe',
        picture: '',
        title: 'mr',
        gender: 'male',
        dateOfBirth: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        registerDate: {
          ruDateAndTime: '05 October 2011 14:48 UTC',
        },
        id: '12345',
        email: '',
        phone: '55555'
      }
    }
  },
  postModalPost: {
    isLoading: false,
    error: false,
    owner: {
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    },
    post: {
      id: '12345',
      image: '',
      text: 'text',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }
  },
  postModalComments: {
    page: 1,
    error: false,
    isLoading: false,
    comments: [{
      id: '1234',
      message: 'message',
      publishDate: {
        ruDateAndTime: '05 October 2011 14:48 UTC'
      }
    }],
    owners: [{
      firstName: 'John',
      lastName: 'Doe',
      picture: '',
      title: 'mr',
      id: '12345'
    }],
    limit: 6,
    totalComments: 0
  }
};

export default initialStore;
