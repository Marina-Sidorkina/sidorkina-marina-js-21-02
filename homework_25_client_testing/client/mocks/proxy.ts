import MockAdapter from 'axios-mock-adapter';
import { proxy } from '../src/api/proxy';

export const mock = new MockAdapter(proxy);
