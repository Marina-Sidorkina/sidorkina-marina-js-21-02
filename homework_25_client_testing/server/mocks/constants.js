module.exports = {
  errorText: 'Request failed with status code 520',
  successStatus: 200,
  errorStatus: 520
};
