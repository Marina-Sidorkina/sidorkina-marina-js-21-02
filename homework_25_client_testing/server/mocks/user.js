module.exports = {
  userListMockData: { data: [], page: 0, limit: 5, total: 100 },
  userMockData: {
    id: '60d0fe4f5311236168a109ce',
    title: 'mr',
    firstName: 'Rudi',
    lastName: 'Droste',
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg',
    gender: 'male',
    email: 'rudi.droste@example.com',
    dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787',
    location: {
      street: '1196, MÃ¼hlenweg',
      city: 'GrÃ¼nhain-Beierfeld',
      state: 'Sachsen-Anhalt',
      country: 'Germany',
      timezone: '+3:30'
    },
    registerDate: '2021-06-21T21:02:08.029Z',
    updatedDate: '2021-06-21T21:02:08.029Z'
  },
  userProcessedMockData: {
    id: '60d0fe4f5311236168a109ce',
    firstName: 'Rudi',
    lastName: 'Droste',
    title: 'mr',
    gender: 'male',
    email: 'rudi.droste@example.com',
    phone: '0421-7342787',
    dateOfBirth: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    registerDate: {
      enDate: '22 of June 2021',
      ruDate: '22 Июня 2021 года',
      enDateAndTime: '22 of June 00:02',
      ruDateAndTime: '22 Июня 00:02'
    },
    picture: 'https://randomuser.me/api/portraits/med/men/83.jpg'
  },
  userPostsListMockData: {
    data: [{id: '12345', image: 'image', owner: {}, text: 'text'}],
    page: 0, limit: 1, total: 100
  },
  userUpdatedMockData: { firstName: 'Rudi', lastname: 'Droste', gender: 'male',
    email: 'rudi.droste@example.com', dateOfBirth: '2021-06-21T21:02:08.029Z',
    phone: '0421-7342787'
  }
};
