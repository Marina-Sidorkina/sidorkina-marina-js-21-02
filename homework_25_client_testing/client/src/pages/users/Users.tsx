import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Main from '../../components/commonComponents/main/Main';
import UsersList from '../../components/usersListComponents/usersList/UsersList';
import Paginator from '../../components/commonComponents/paginator/Paginator';
import { updateUsersListPageAction } from '../../redux/actions/usersList';
import { IUsersProps } from './@types/users';
import { DEFAULT_PAGE } from '../../constants/components';

const Users = (props: IUsersProps) => {
  const {
    page, total, perPage, updatePage
  } = props;

  useEffect(() => () => updatePage(DEFAULT_PAGE), []);

  return (
    <Main>
      <UsersList />
      <Paginator
        current={page}
        total={total}
        perPage={perPage}
        onPageChange={(value: number) => {
          updatePage(value);
        }}
      />
    </Main>
  );
};

export default connect(
  (state: any) => ({
    page: state.usersList.data.page,
    total: state.usersList.data.total,
    perPage: state.usersList.data.perPage,
  }),
  (dispatch) => ({
    updatePage: bindActionCreators(updateUsersListPageAction, dispatch)
  })
)(Users);
