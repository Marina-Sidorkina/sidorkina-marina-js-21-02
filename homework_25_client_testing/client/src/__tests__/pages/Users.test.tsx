import '../../../mocks/matchMediaMock';
import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import Users from '../../pages/users/Users';
import mockStore from '../../../mocks/mockStore';
import initialStore from '../../../mocks/initialStore';

configure({ adapter: new Adapter() });

describe('Users test', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = mockStore(initialStore);
    wrapper = mount(
      <Provider store={store}>
        <HashRouter>
          <Users />
        </HashRouter>
      </Provider>
    );
  });

  test('should render Users', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('should render Users', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
