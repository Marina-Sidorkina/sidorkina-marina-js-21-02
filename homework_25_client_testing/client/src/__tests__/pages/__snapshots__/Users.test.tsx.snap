// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users test should render Users 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Users)>
        <Users
          page={0}
          perPage={6}
          total={100}
          updatePage={[Function]}
        >
          <Main>
            <main
              className="main"
            >
              <div
                className="container"
              >
                <UsersList>
                  <ul
                    className="list"
                  >
                    <UsersListItem
                      firstName="John"
                      id="12345"
                      key="12345"
                      lastName="Doe"
                      title="mr"
                    >
                      <Link
                        className="link"
                        to="/profile/12345"
                      >
                        <LinkAnchor
                          className="link"
                          href="#/profile/12345"
                          navigate={[Function]}
                        >
                          <a
                            className="link"
                            href="#/profile/12345"
                            onClick={[Function]}
                          >
                            <li
                              className="item"
                            >
                              <img
                                alt="Аватар пользователя"
                                className="img"
                                src="https://i.ibb.co/2yWbJxP/default.png"
                              />
                              <ComponentWithHelper
                                firstName="John"
                                lastName="Doe"
                                title="mr"
                              >
                                <div
                                  className="helper"
                                  onMouseOut={[Function]}
                                  onMouseOver={[Function]}
                                >
                                  <UsersListItemName
                                    firstName="John"
                                    lastName="Doe"
                                    title="mr"
                                  >
                                    <div>
                                      г-н John Doe
                                    </div>
                                  </UsersListItemName>
                                </div>
                              </ComponentWithHelper>
                            </li>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </UsersListItem>
                  </ul>
                </UsersList>
                <Paginator
                  current={0}
                  modal={false}
                  onPageChange={[Function]}
                  perPage={6}
                  total={100}
                >
                  <Pagination
                    className="paginator
        "
                    current={0}
                    onChange={[Function]}
                    pageSize={6}
                    showQuickJumper={true}
                    simple={true}
                    total={100}
                  >
                    <LocaleReceiver
                      componentName="Pagination"
                      defaultLocale={
                        Object {
                          "items_per_page": "/ page",
                          "jump_to": "Go to",
                          "jump_to_confirm": "confirm",
                          "next_3": "Next 3 Pages",
                          "next_5": "Next 5 Pages",
                          "next_page": "Next Page",
                          "page": "Page",
                          "page_size": "Page Size",
                          "prev_3": "Previous 3 Pages",
                          "prev_5": "Previous 5 Pages",
                          "prev_page": "Previous Page",
                        }
                      }
                    >
                      <Pagination
                        className="paginator
        "
                        current={0}
                        defaultCurrent={1}
                        defaultPageSize={10}
                        hideOnSinglePage={false}
                        itemRender={[Function]}
                        jumpNextIcon={
                          <a
                            className="ant-pagination-item-link"
                          >
                            <div
                              className="ant-pagination-item-container"
                            >
                              <ForwardRef(DoubleRightOutlined)
                                className="ant-pagination-item-link-icon"
                              />
                              <span
                                className="ant-pagination-item-ellipsis"
                              >
                                •••
                              </span>
                            </div>
                          </a>
                        }
                        jumpPrevIcon={
                          <a
                            className="ant-pagination-item-link"
                          >
                            <div
                              className="ant-pagination-item-container"
                            >
                              <ForwardRef(DoubleLeftOutlined)
                                className="ant-pagination-item-link-icon"
                              />
                              <span
                                className="ant-pagination-item-ellipsis"
                              >
                                •••
                              </span>
                            </div>
                          </a>
                        }
                        locale={
                          Object {
                            "items_per_page": "/ page",
                            "jump_to": "Go to",
                            "jump_to_confirm": "confirm",
                            "next_3": "Next 3 Pages",
                            "next_5": "Next 5 Pages",
                            "next_page": "Next Page",
                            "page": "Page",
                            "page_size": "Page Size",
                            "prev_3": "Previous 3 Pages",
                            "prev_5": "Previous 5 Pages",
                            "prev_page": "Previous Page",
                          }
                        }
                        nextIcon={
                          <button
                            className="ant-pagination-item-link"
                            tabIndex={-1}
                            type="button"
                          >
                            <ForwardRef(RightOutlined) />
                          </button>
                        }
                        onChange={[Function]}
                        onShowSizeChange={[Function]}
                        pageSize={6}
                        prefixCls="ant-pagination"
                        prevIcon={
                          <button
                            className="ant-pagination-item-link"
                            tabIndex={-1}
                            type="button"
                          >
                            <ForwardRef(LeftOutlined) />
                          </button>
                        }
                        selectComponentClass={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "OptGroup": [Function],
                            "Option": [Function],
                            "SECRET_COMBOBOX_MODE_DO_NOT_USE": "SECRET_COMBOBOX_MODE_DO_NOT_USE",
                            "render": [Function],
                          }
                        }
                        selectPrefixCls="ant-select"
                        showLessItems={false}
                        showPrevNextJumpers={true}
                        showQuickJumper={true}
                        showTitle={true}
                        simple={true}
                        style={Object {}}
                        total={100}
                        totalBoundaryShowSizeChanger={50}
                      >
                        <ul
                          className="ant-pagination ant-pagination-simple paginator
        "
                          style={Object {}}
                        >
                          <li
                            aria-disabled={true}
                            className="ant-pagination-prev ant-pagination-disabled"
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            tabIndex={null}
                            title="Previous Page"
                          >
                            <button
                              className="ant-pagination-item-link"
                              disabled={true}
                              tabIndex={-1}
                              type="button"
                            >
                              <ForwardRef(LeftOutlined)>
                                <AntdIcon
                                  icon={
                                    Object {
                                      "icon": Object {
                                        "attrs": Object {
                                          "focusable": "false",
                                          "viewBox": "64 64 896 896",
                                        },
                                        "children": Array [
                                          Object {
                                            "attrs": Object {
                                              "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                            },
                                            "tag": "path",
                                          },
                                        ],
                                        "tag": "svg",
                                      },
                                      "name": "left",
                                      "theme": "outlined",
                                    }
                                  }
                                >
                                  <span
                                    aria-label="left"
                                    className="anticon anticon-left"
                                    role="img"
                                  >
                                    <IconReact
                                      icon={
                                        Object {
                                          "icon": Object {
                                            "attrs": Object {
                                              "focusable": "false",
                                              "viewBox": "64 64 896 896",
                                            },
                                            "children": Array [
                                              Object {
                                                "attrs": Object {
                                                  "d": "M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z",
                                                },
                                                "tag": "path",
                                              },
                                            ],
                                            "tag": "svg",
                                          },
                                          "name": "left",
                                          "theme": "outlined",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="left"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        key="svg-left"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                          key="svg-left-svg-0"
                                        />
                                      </svg>
                                    </IconReact>
                                  </span>
                                </AntdIcon>
                              </ForwardRef(LeftOutlined)>
                            </button>
                          </li>
                          <li
                            className="ant-pagination-simple-pager"
                            title="0/17"
                          >
                            <input
                              onBlur={[Function]}
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              size="3"
                              type="text"
                              value={0}
                            />
                            <span
                              className="ant-pagination-slash"
                            >
                              /
                            </span>
                            17
                          </li>
                          <li
                            aria-disabled={false}
                            className="ant-pagination-next"
                            onClick={[Function]}
                            onKeyPress={[Function]}
                            tabIndex={null}
                            title="Next Page"
                          >
                            <button
                              className="ant-pagination-item-link"
                              disabled={false}
                              tabIndex={-1}
                              type="button"
                            >
                              <ForwardRef(RightOutlined)>
                                <AntdIcon
                                  icon={
                                    Object {
                                      "icon": Object {
                                        "attrs": Object {
                                          "focusable": "false",
                                          "viewBox": "64 64 896 896",
                                        },
                                        "children": Array [
                                          Object {
                                            "attrs": Object {
                                              "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                            },
                                            "tag": "path",
                                          },
                                        ],
                                        "tag": "svg",
                                      },
                                      "name": "right",
                                      "theme": "outlined",
                                    }
                                  }
                                >
                                  <span
                                    aria-label="right"
                                    className="anticon anticon-right"
                                    role="img"
                                  >
                                    <IconReact
                                      icon={
                                        Object {
                                          "icon": Object {
                                            "attrs": Object {
                                              "focusable": "false",
                                              "viewBox": "64 64 896 896",
                                            },
                                            "children": Array [
                                              Object {
                                                "attrs": Object {
                                                  "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                                                },
                                                "tag": "path",
                                              },
                                            ],
                                            "tag": "svg",
                                          },
                                          "name": "right",
                                          "theme": "outlined",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="right"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        key="svg-right"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                                          key="svg-right-svg-0"
                                        />
                                      </svg>
                                    </IconReact>
                                  </span>
                                </AntdIcon>
                              </ForwardRef(RightOutlined)>
                            </button>
                          </li>
                        </ul>
                      </Pagination>
                    </LocaleReceiver>
                  </Pagination>
                </Paginator>
              </div>
            </main>
          </Main>
        </Users>
      </Connect(Users)>
    </Router>
  </HashRouter>
</Provider>
`;
